packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -87,9 +89,13 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   Future<T> get completed => _transitionCompleter?.future;
   final Completer<T> _transitionCompleter;
 
+  /// The duration the transition lasts.
   Duration get transitionDuration;
+
+  /// Whether the route obscures previous routes when the transition is complete.
```

Comment:
add something about how if this is true, we discard the previous routes to save memory etc